File: ValidParentheses_20/solution.ts
function isValid(s: string): boolean {
    const stack: string[] = [];
    
    for (const char of s) {
      if (char === '(') {
        stack.push(')');
        continue;
      }
      
      if (char === '{') {
        stack.push('}');
        continue;
      } 
      
      if (char === '[') {
        stack.push(']');
        continue;
      } 
      
      if (stack.length === 0 || char !== stack.pop()) {
        return false;
      }
    }

    return stack.length === 0;
}

File: ValidParentheses_20/solution.cs
public class Solution {
    public bool IsValid(string s) {
        var stack = new Stack<char>();

        foreach (var c in s)
        {
            if( c == '(') 
            {
                stack.Push(')'); 
                continue;
            }

            if( c == '{') 
            { 
                stack.Push('}'); 
                continue;
            }

            if( c == '[') 
            { 
                stack.Push(']'); 
                continue;
            }

            if(stack.Count == 0 || c != stack.Pop()) return false;
        }
    
        return stack.Count == 0;
    }
}